### PR TITLE
Updated code to remove dependency on IS_LOADING_POLICY_DATA for Welcome.js

### PR DESCRIPTION
**Pull request by techievivek**
### Details

In Welcome.js remove the usage of `IS_LOADING_POLICY_DATA` because report and policy data are loaded together so we can just rely on `IS_LOADING_REPORT_DATA` to trust that both report and policy data are available.
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/211216   
PROPOSAL: GH_LINK_ISSUE(COMMENT)

### Tests

1. Create a new workspace from an account that's on the policyExpenseChat beta (e.g @expensifail.com) since for DEV every beta is enabled by default we can use any account.
2. To this newly created workspace added a member by going to Manage Members `>` Invite.
3. CMD+F and search the member's name to navigate to their workspace chat. (It could be that you will not find the workspace chat so you can just refresh the app and that should load the workspace chat). 
The workspace chat would look something like this.

Screenshot 2022-10-06 at 3 11 02 PM

4. Write a couple of messages in the member's workspace chat, then log out of the admin account.
5. Validated the newly added member account and logged in to NewDot. On DEV you will use `./clitools.sh` on PROD you can click on the email.
6. We now land in the workspace chat and we are able to see the messages sent by the admin.

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Create a new workspace from an account that's on the policyExpenseChat beta (e.g @expensifail.com).
2. To this newly created workspace added a member by going to Manage Members `>` Invite.
3. CMD+F and search the member's name to navigate to their workspace chat. (It could be that you will not find the workspace chat so you can just refresh the app and that should load the workspace chat). 
The workspace chat would look something like this.

Screenshot 2022-10-06 at 3 11 02 PM

4. Write a couple of messages in the member's workspace chat, then log out of the admin account.
5. Validated the newly added member account and log in to NewDot.
6. Make sure you land in the workspace chat.
- [ ] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/35863227/194282917-57718d63-91b0-4f99-aa8f-d853420d6883.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/35863227/194834544-f1440a9d-5f3c-4bbb-ab77-a458581c192f.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/35863227/194836313-f81d05cb-a72b-452e-92f8-78ef8897121e.mov

#### Desktop

https://user-images.githubusercontent.com/35863227/194834572-87856e4a-4cba-434d-ac30-2199c366025e.mov

#### iOS

https://user-images.githubusercontent.com/35863227/194834582-23ae41cc-cf06-4193-95d1-3cdd12c3b2be.mov

#### Android

https://user-images.githubusercontent.com/35863227/194834601-3a390b7b-362a-4692-85fe-34c58bf8d197.mov

### CODE REVIEW

**Review by luacmartins — CHANGES_REQUESTED** [review]
We are intentionally removing the `IS_LOADING_POLICY_DATA` key. Instead of adding it back, I'd suggest that we remove the Welcome event's dependency on it.

File: src/libs/actions/App.js [hunk 1]
```diff
@@ -121,17 +121,20 @@ function openApp() {
             onyxMethod: CONST.ONYX.METHOD.MERGE,
             key: ONYXKEYS.IS_LOADING_REPORT_DATA,
             value: true,
-        }],
+        },
+        ],
```
Comment (bondydaa) [inline src/libs/actions/App.js:125]: strange I thought we'd prefer what was there before (`}]`) did the linter complain about this?

**Review by bondydaa — COMMENTED** [review]
Looks like the only other spots we use that Onyx key is here https://github.com/Expensify/App/blob/ec7d84b4e3411f6aa08e3d189177216f1b5c53e9/src/libs/actions/Policy.js#L183-L203 so we can we just fully kill this key and code?

File: src/libs/actions/App.js (hunk 1 above)
Comment (luacmartins) [reply src/libs/actions/App.js:125]: +1 I think the other way reads better. Same for the others.

**Review by luacmartins — CHANGES_REQUESTED** [review]
I agree with Bondy's comments. Let's revert the style changes in `actions/App` and remove [this code](https://github.com/Expensify/App/blob/6406b54c8f3b414ab80e05a8aa03f17d5068fb01/src/libs/actions/Policy.js#L150-L182) and [this Onyx key](https://github.com/Expensify/App/blob/6406b54c8f3b414ab80e05a8aa03f17d5068fb01/src/ONYXKEYS.js#L158-L159)

**Review by bondydaa — DISMISSED** [review]
👍 nice work.

**Review by luacmartins — COMMENTED** [review]
Looking good! While reviewing this, I noticed that we might have another bug (not introduced in this PR)

This [code](https://github.com/Expensify/App/blob/6406b54c8f3b414ab80e05a8aa03f17d5068fb01/src/libs/actions/Welcome.js#L68-L70) is supposed to load all policies, but it actually loads all reports. So the first part of [this conditional](https://github.com/Expensify/App/blob/6406b54c8f3b414ab80e05a8aa03f17d5068fb01/src/libs/actions/Welcome.js#L123) will probably always be true and admins of workspaces would also see the welcome action.

We can fix that in this PR too!

**Comment by techievivek** [comment]
> So the first part of [this conditional](https://github.com/Expensify/App/blob/6406b54c8f3b414ab80e05a8aa03f17d5068fb01/src/libs/actions/Welcome.js#L123) will probably always be true and admins of workspaces would also see the welcome action

Great catch, since policy data are not loaded this makes sense.

**Review by luacmartins — COMMENTED** [review]
@techievivek I had to merge main to be able to see the policy expense chats. The solution seem to work well on web, desktop and mWeb Chrome, but it did not work on android, iOS or mWeb safari.

I see that you had videos with the solution working, so I'm wondering if it's just me holding something wrong. @bondydaa could you try to test this solution in iOS/android?


https://user-images.githubusercontent.com/22219519/195633806-7517fb80-c295-47df-ac35-307d7a22a339.mov


https://user-images.githubusercontent.com/22219519/195633810-6190c5f7-07e8-4f39-ae08-2dfe9d0fb9cd.mov


https://user-images.githubusercontent.com/22219519/195633812-90e8014f-b98a-4790-a2cd-1264ad0e964d.mov


https://user-images.githubusercontent.com/22219519/195633816-0f3e5976-f265-4a2f-9e55-a2de59561597.mov


https://user-images.githubusercontent.com/22219519/195633820-88ed85ac-34f2-41e8-b217-50126c4baeca.mov


https://user-images.githubusercontent.com/22219519/195633821-7b81f405-fa68-47b8-af80-c2261bb9b5e3.mov

**Comment by techievivek** [comment]
That's odd, let me try testing it once again and probably adding some log statements to make sure logic is being hit.

**Comment by techievivek** [comment]
I struggled with the DEV environment setup yesterday, so couldn't look into it. I will test it today.

**Comment by techievivek** [comment]
I tested this locally and experienced the same thing that @luacmartins experienced. I wasn't able to test this further because of the `Pusher limit` which didn't allow me to create new accounts on DEV. But this is something that we will have to make sure gets resolved before moving forward.

In the logs, it does show that we will be navigation to the workspace chat but it just freezes.

Screenshot 2022-10-18 at 3 59 01 PM

**Comment by luacmartins** [comment]
@techievivek since this is a react navigation issue, I think we should merge this to partially fix the problem and then open another issue to investigate the RN nav issue. Could you please update your branch with main? I'll review it again and merge this PR then.

**Comment by techievivek** [comment]
Alright, I have merged with main now.

**Comment by luacmartins** [comment]
@techievivek can you update your author checklist? I think that we have a new version and that's why your PR Author checklist check is failing

**Comment by techievivek** [comment]
@luacmartins I have ticked all the options. I am not sure why it is still failing.

**Comment by techievivek** [comment]
@luacmartins, thanks for the note, I have updated the checklists.

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

https://user-images.githubusercontent.com/22219519/195633806-7517fb80-c295-47df-ac35-307d7a22a339.mov

https://user-images.githubusercontent.com/22219519/195633810-6190c5f7-07e8-4f39-ae08-2dfe9d0fb9cd.mov

https://user-images.githubusercontent.com/22219519/195633812-90e8014f-b98a-4790-a2cd-1264ad0e964d.mov

https://user-images.githubusercontent.com/22219519/195633816-0f3e5976-f265-4a2f-9e55-a2de59561597.mov

https://user-images.githubusercontent.com/22219519/195633820-88ed85ac-34f2-41e8-b217-50126c4baeca.mov

https://user-images.githubusercontent.com/22219519/195633821-7b81f405-fa68-47b8-af80-c2261bb9b5e3.mov

**Comment by luacmartins** [comment]
@techievivek I created a [new issue](https://github.com/Expensify/App/issues/12041) for the react navigation quirk on iOS and android and assigned it to you.

**Comment by marcaaron** [comment]
>In Welcome.js remove the usage of IS_LOADING_POLICY_DATA because report and policy data are loaded together so we can just rely on IS_LOADING_REPORT_DATA to trust that both report and policy data are available.

Late to the party, but I would have re-named this if we are relaying on a "report data is loading" flag to mean more than just the report data is loading.

**Pull request merged**